zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/CommitProcessor.java

```diff
@@ -200,13 +202,9 @@ public void run() {
                     if (needCommit(request)
                             || pendingRequests.containsKey(request.sessionId)) {
                         // Add request to pending
-                        LinkedList<Request> requests = pendingRequests
-                                .get(request.sessionId);
-                        if (requests == null) {
-                            requests = new LinkedList<Request>();
-                            pendingRequests.put(request.sessionId, requests);
-                        }
-                        requests.addLast(request);
+                        pendingRequests
+                            .computeIfAbsent(request.sessionId, sid -> new ArrayDeque<>())
+                            .add(request);
```

Comment:
FYI.  This is exactly how this is intended to be used:

https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-